tools/resourcedocsgen/cmd/metadata.go

```diff
@@ -649,6 +656,15 @@ func inferPublishDate(repo repoSlug, version string) (time.Time, bool, error) {
 	return commit.Commit.Author.Date, true, nil
 }
 
+func computeEditURLFromGitHubUserContentURL(url string) *string {
+	parts := strings.Split(url, "/")
+	if len(parts) < 6 {
+		return nil
+	}
+	result := "https://github.com/" + parts[3] + "/" + parts[4] + "/blob/" + strings.Join(parts[5:], "/")
+	return &result
```

Comment:
Idiomatic Go would have this return an empty string for no result found.
```suggestion
func computeEditURLFromGitHubUserContentURL(url string) string {
	parts := strings.Split(url, "/")
	if len(parts) < 6 {
		return nil
	}
	return "https://github.com/" + parts[3] + "/" + parts[4] + "/blob/" + strings.Join(parts[5:], "/")
```